storage/redis_cluster.go

```diff
@@ -159,11 +156,90 @@ func NewRedisClusterPool(isCache bool) redis.UniversalClient {
 		IdleTimeout:  240 * timeout,
 		PoolSize:     poolSize,
 		TLSConfig:    tlsConfig,
-	})
+	}
+
+	if cfg.EnableCluster {
+		log.Info("--> [REDIS] Using clustered mode")
+		client = redis.NewClusterClient(opts.cluster())
+	} else {
+		log.Info("--> [REDIS] Using single node mode")
+		client = redis.NewClient(opts.simple())
```

Comment:
As mentioned previously, lets then add smth like:
``
	else if opts.MasterName != "" {
		return NewFailoverClient(opts.failover())
       }
```

Reply:
As discussed in Slack will be separate PR